### PR TITLE
Feature/micromamba

### PR DESCRIPTION
**Pull request by ahirner**
This PR fixes failing builds and makes them simpler.

### Changes:
- Switch `python-base` base images to [micromamba-docker](https://github.com/mamba-org/micromamba-docker)
  - Upgrades base images indirectly to debian 11.5 bullseye
  - Replaces heritage of little maintained  [phusion](https://github.com/phusion/baseimage-docker) with [runit-docker](https://github.com/wk8/docker-runit) installation only
- Remove nvidia base images in favour of cuda runtime packaged in pytorch wheels
- `ffmpeg` and `torch` builds in `python-base` image for less compatibility risks
- Squash `.deb` and `.whl` packages in `moonbox` images
- Total compressed size savings for `cuda-latest`: 3.71 to 2.2GB
- Remove obsolete `cmake` build since appropriate version is available in new base image

- Fix deprecated github action statements

### Todos:
- [x] Remove stale `docker` code
- [x] Update `Readme.md` and dependency graph
- [x] Ensure downstream build compatibility
- [x] Test downstream deploy compatibilities

### How to build locally:

In contrast to `Readme.md`, all default `build` arguments that reference `moonvision` images, need to be overridden with the `dev-` prefix until this PR is merged.

E.g. to build moonbox-mini:
```
docker build  -t moonvision/dev-moonbox:mini-latest \
  --build-arg baseimage=moonvision/dev-python-base \
  --build-arg ffmpeg_from_docker=moonvision/dev-custom-builds:ffmpeg-4.2.1 \
  --build-arg pytorch_from_docker=moonvision/dev-custom-builds:pytorch-1.11.0_torchvision-0.12.0 \
  -f docker/moonbox/Dockerfile docker/moonbox
```

### CODE REVIEW

**Comment by truh** [comment]
```
docker build \
  -t moonvision/moonbox:mini-latest \
  -f docker/moonbox/Dockerfile docker/moonbox
```

gives me 

Screen Shot 2023-01-11 at 10 58 29

File: Readme.md [hunk 1]
```diff
@@ -8,189 +8,137 @@ Docker images for computer vision and deep learning.
...
+| `baseimage` | `mambaorg/micromamba:1` | Set the image in which FFmpeg will be built. |
+| `ffmpeg_version` | | The FFmpeg release version to build. |
+| `with_cuda` | `false` | Whether to build FFmpeg with Nvidia hardware acceleration.  |
 
-#### FFmpeg
+#### FFmpeg without CUDA
 
 ```
-docker build -t moonvision/custom-builds:ffmpeg-latest -f docker/builders/ffmpeg/Dockerfile --build-arg ffmpeg_version=4.2.1 docker/builders/ffmpeg
+docker build \
```
Comment (truh) [inline Readme.md:102]: There appears to be some weird special character or something in this command, it doesn't parse correctly.

Screen Shot 2023-01-11 at 11 06 48

Comment (truh) [reply Readme.md:102]: Or maybe a space after one of the backslashes

File: Readme.md [hunk 2]
```diff
@@ -8,189 +8,137 @@ Docker images for computer vision and deep learning.
...
-| `ffmpeg_version` | `4.2.1` | The FFmpeg release version to build. |
-| `with_cuda` | `false` | Whether to build FFmpeg with Nvidia hardware acceleration.  |
-
 #### FFmpeg with CUDA
 
 ```
-docker build -t moonvision/custom-builds:ffmpeg-cuda-latest -f docker/builders/ffmpeg/Dockerfile --build-arg baseimage=nvidia/cuda:11.0.3-base-ubuntu20.04 --build-arg ffmpeg_version=4.2.1 --build-arg with_cuda=true docker/builders/ffmpeg
+docker build \
+  -t moonvision/custom-builds:ffmpeg-cuda-latest \
+  --build-arg ffmpeg_version=4.2.1 \
```
Comment (truh) [inline Readme.md]:
```suggestion
  --build-arg ffmpeg_version=4.2.1 \
```

File: Readme.md [hunk 3]
```diff
@@ -8,189 +8,137 @@ Docker images for computer vision and deep learning.
...
+| `with_cuda` | `false` | Whether to build FFmpeg with Nvidia hardware acceleration.  |
 
-#### FFmpeg
+#### FFmpeg without CUDA
 
 ```
-docker build -t moonvision/custom-builds:ffmpeg-latest -f docker/builders/ffmpeg/Dockerfile --build-arg ffmpeg_version=4.2.1 docker/builders/ffmpeg
+docker build \
+  -t moonvision/custom-builds:ffmpeg-latest \
+  --build-arg ffmpeg_version=4.2.1 \
```
Comment (truh) [inline Readme.md]:
```suggestion
  --build-arg ffmpeg_version=4.2.1 \
```

File: Readme.md [hunk 4]
```diff
@@ -8,189 +8,137 @@ Docker images for computer vision and deep learning.
...
-| `python_from_docker` | `moonvision/python-base:cuda-latest` | Image to use conda python from for building. |
-| `pytorch_tag` | `v1.11.0` | PyTorch version (tag or commit if built, pip version if downloaded). |
-| `torchvision_tag` | `v0.12.0` | Torchvision version (tag or commit if built, pip version if downloaded). |
-| `prebuilt` | `true` | If `true` [download](https://pytorch.org/get-started/locally/) pip versions instead of building from source. |
+#### PyTorch with CUDA
+
+```
+docker build \
+  -t moonvision/custom-builds:pytorch-cuda-latest \
+  --build-arg with_cuda=true \
```
Comment (truh) [inline Readme.md]:
```suggestion
  --build-arg with_cuda=true \
```

**Review by truh — CHANGES_REQUESTED** [review]
that's it for now

**Comment by ahirner** [comment]
> gives me [missing dll]

Please try this and see updated PR description for why:
```
docker build  -t moonvision/dev-moonbox:mini-latest \
  --build-arg baseimage=moonvision/dev-python-base \
  --build-arg ffmpeg_from_docker=moonvision/dev-custom-builds:ffmpeg-4.2.1 \
  --build-arg pytorch_from_docker=moonvision/dev-custom-builds:pytorch-1.11.0_torchvision-0.12.0 \
  -f docker/moonbox/Dockerfile docker/moonbox
```

File: Readme.md (hunk 1 above)
Comment (ahirner) [reply Readme.md:102]: Thanks, indeed a space invaded the commands. Should work now.

**Review by truh — APPROVED** [review]
Good  work.

**Pull request merged**